### PR TITLE
feat: WDS color `bgAssistive`

### LINKED ISSUE #23986: feat: WDS assistive colors algorithm
Currently we're [simply cloning fg into bgAssistive](https://github.com/appsmithorg/appsmith/blob/eafe34244a728fd17a69d9a149c8ecbe2c246db4/app/client/packages/design-system/theming/src/utils/TokensAccessor/LightModeTheme.ts#L219), should be properly defined instead.

### PR DESCRIPTION
**Pull request by ichik**
## Description

tl;dr Generative backgrounds for tooltips etc.

Fixes #23986

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Manual

### Test Plan
Initial POC refinement, no testing necessary

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/packages/design-system/theming/src/color/DarkModeTheme.ts [hunk 1]
```diff
@@ -546,7 +546,17 @@ export class DarkModeTheme implements ColorModeTheme {
   }
 
   private get bgAssistive() {
-    return this.fg.clone();
+    const color = this.seedColor.clone();
```
Comment (KelvinOm) [inline app/client/packages/design-system/theming/src/color/DarkModeTheme.ts:549]: Is it considered an edge case when we have a very dark seed or very light in dark mode, then the tooltip looks exactly the same? Do we want to do something about it?
Снимок экрана 2023-07-18 в 18 28 50
Снимок экрана 2023-07-18 в 18 28 40

**Review by KelvinOm — APPROVED** [review]

File: app/client/packages/design-system/theming/src/color/DarkModeTheme.ts (hunk 1 above)
Comment (KelvinOm) [reply app/client/packages/design-system/theming/src/color/DarkModeTheme.ts:549]: It looks like a ticket for ice box.
